freebites/product
# 103 signup login keyboard glitch

# Description

Fixed a number of styling issues on the login and signup page including: fixed forgot password and remember me spacing, made sure the correct errors displayed when trying to log in, fixing cut off "terms of use" line, fixed and added scrolling capabilities to the keyboard avoiding view when entering things into the text input fields. Fixed the glitchy effect when switching between text inputs in the keyboard view

## Issues
Resolve #103 


## Screenshots

<img src="https://github.com/user-attachments/assets/f389f9da-bab6-46cf-b430-7ed65998fcc5" width=250>
<img src="https://github.com/user-attachments/assets/533892e0-f134-4708-b83f-d5f9a0274304" width=250>
<img src="https://github.com/user-attachments/assets/fa3cd6a6-60e6-4cb3-9171-72cb514ae506" width=250>
<img src="https://github.com/user-attachments/assets/4067089d-a2b2-4f8a-bae9-8286958e594b" width=250>
<img src="https://github.com/user-attachments/assets/c5971432-7db5-4883-91e9-1015e1e8ebc9" width=250>
<img src="https://github.com/user-attachments/assets/79aef5ef-109f-4bc1-b6ae-4de977842c86" width=250>

## Test

- open the app and log out if automatically logged in
- tap login, the page should match the figma
- type in a bad email, should get email error
- type in a good email and a bad password and should get the password error
- while typing in these fields you should have no issues with the keyboard avoiding view covering up anything
- go back and tap sign up, the page should match the figma (specifically should see the bottom line of the terms and conditions statement)
- when tapping on one of the top fields the keyboard should appear without moving the ui elements
- tap on elements lower down like username and password both while the keyboard is already active and when it is not active
- try scrolling through the inputs while the keyboard is open

## Possible Downsides

to get the proper email/password errors to display I had to override the existing error display system a bit, so it is possible that the incorrect password error may display in cases where something else went wrong during the login (though I think previously it would have just shown the email error in that case anyway). Had very minor ui scrolling issues on the login page but have not been able to replicate them in recent testing despite not having intentionally fixed anything.

## Additional Documentations

added a new npm package: https://www.npmjs.com/package/react-native-keyboard-aware-scroll-view
